diff --git a/third_party/xla/xla/backends/cpu/runtime/convolution_thunk_internal.h b/third_party/xla/xla/backends/cpu/runtime/convolution_thunk_internal.h
index 84fed6bb78657..7e22492e7857b 100644
--- a/third_party/xla/xla/backends/cpu/runtime/convolution_thunk_internal.h
+++ b/third_party/xla/xla/backends/cpu/runtime/convolution_thunk_internal.h
@@ -338,15 +338,16 @@ void EigenGenericConv2D(
     auto num_tasks = Eigen::numext::div_ceil(feature_group_count, task_size);
 
     if (use_thunk_runtime) {
-      ScheduleAll(&device, num_tasks, [=, &device](Eigen::Index task_index) {
-        Eigen::Index start = task_index * task_size;
-        Eigen::Index end = std::min(start + task_size, feature_group_count);
-        for (Eigen::Index i = start; i < end; ++i) {
-          auto on_done = [count_down]() mutable { count_down.CountDown(); };
-          auto [output, convolved] = convolve_group(i);
-          output.device(device, std::move(on_done)) = convolved;
-        }
-      });
+      ScheduleAll(
+          &device, num_tasks, [=, &device](Eigen::Index task_index) mutable {
+            Eigen::Index start = task_index * task_size;
+            Eigen::Index end = std::min(start + task_size, feature_group_count);
+            for (Eigen::Index i = start; i < end; ++i) {
+              auto on_done = [count_down]() mutable { count_down.CountDown(); };
+              auto [output, convolved] = convolve_group(i);
+              output.device(device, std::move(on_done)) = convolved;
+            }
+          });
     } else {
       Eigen::Barrier barrier(num_tasks);
       ScheduleAll(
